File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-3.java
class Solution {
    public int search(int[] nums, int target) {
        int i=0,n=nums.length,j=n-1;
        while(j>=i){
            int m=i+(j-i)/2;
            System.out.println(i+" d "+j);
           if(nums[m]==target)return m;
            if(nums[i]<=nums[m]){
                if(target > nums[m] || target < nums[i] ){
                    i=m+1;
                }
                else
                    j=m-1;
            }
            else
                {
                    if(target < nums[m] || target > nums[j])
                    j=m-1;
                    else 
                    i=m+1;
                }
        }
        return -1;
    }
}

//6
// 3 4 5 6 7 1 2
    //    m

File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        dic = defaultdict(int)
        for i in nums:
            dic[i] +=1
        heap = []
        for i in dic.keys():
            heapq.heappush(heap, (dic[i], i))
            if len(heap) > k:
                heapq.heappop(heap)
        res = []
        print(heap)
        for i in range(k):
            res.append(heap[i][1])
        return res